### PR TITLE
Added GitHub PR Workflow Check: Talawa Admin app starts (No Docker)

### LINKED ISSUE #2401: GitHub PR Workflow Check: Talawa Admin app starts `(No Docker)`
**Describe the bug**

The GitHub PR workflow does not check whether the Talawa Admin app can start

**To Reproduce**

1. Make a change to the repo that makes the Talawa Admin app fail to start
1. Trigger the workflow to run as part of a PR
1. The workflow **_passes_**

and as expected:

1. Make a change to the repo that makes the Talawa Admin app start
1. Trigger the workflow to run as part of a PR
1. The workflow passes

**Expected behavior**

1. Make a change to the repo that makes the Talawa Admin app fail to start
1. Trigger the workflow to run as part of a PR
1. The workflow **_fails_**

and as expected:

1. Make a change to the repo that makes the Talawa Admin app start
1. Trigger the workflow to run as part of a PR
1. The workflow passes

We now have a test data that can be loaded that resides in the repo. This should be used as part of the test.

**Actual behavior**

- The Talawa Admin app isn't started and tested in the PR workflow

**Screenshots**

- N/A

**Additional details**

1. It must verify starting without Docker
1. Use this issue for reference:
    1. https://github.com/PalisadoesFoundation/talawa-api/issues/1343

**Potential internship candidates**

- Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by adithyanotfound**
**What kind of change does this PR introduce?**

feature, refactoring

**Issue Number:**

Fixes #2401 

**Did you add tests for your changes?**

Not required.

**If relevant, did you update the documentation?**

Not required.

**Summary**

Added PR workflow to test if app starts.

**Does this PR introduce a breaking change?**

No.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Introduced a new job in the CI workflow to check if the Talawa Admin application starts without Docker.

- **Improvements**
	- Enhanced the continuous integration process by adding health checks for both production and development applications, ensuring proper startup verification.

### CODE REVIEW

**Comment by palisadoes** [comment]
Please modify your submission so that coderabbit approves the PR

**Comment by adithyanotfound** [comment]
@coderabbitai review

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -260,6 +260,85 @@ jobs:
...
+    if: github.actor != 'dependabot'
+    steps:
+      - name: Checkout the Repository
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+
```
[the remark this answers is not in the document]
Comment (adithyanotfound) [reply .github/workflows/pull-request.yml:276]: Not required.

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -260,6 +260,85 @@ jobs:
...
+              echo "Still waiting for development app to start... ${timeout}s remaining"
+            fi
+          done
+
+          if [ $timeout -eq 0 ]; then
+            echo "Timeout waiting for development application to start"
+            exit 1
+          fi
+          echo "Development app started successfully"
+
```
[the remark this answers is not in the document]
Comment (adithyanotfound) [reply .github/workflows/pull-request.yml:334]: Not valid.

**Comment by adithyanotfound** [comment]
> Please modify your submission so that coderabbit approves the PR

@palisadoes I have made the required changes.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**